src/specBuilder/signal/signalSpecBuilder.ts

```diff
@@ -121,3 +147,151 @@ export const addHighlightedSeriesSignalEvents = (signals: Signal[], markName: st
 		);
 	}
 };
+/**
+ * get all signals required for opacity animations
+ * @param name
+ * @param nestedDatum
+ * @param isNull
+ * @returns Signal[]
+ */
+//TODO: Add tests
+export const getRscAnimationSignals = (name: string, nestedDatum?: boolean, isNull?: boolean): Signal[] => {
+	return [
+		getRscAnimation(),
+		getRscColorAnimationDirection(name),
+		getRscColorAnimation(),
+		getRscHighlightedItemPrevSignal(name, nestedDatum),
+		getRscHighlightedSeriesPrevSignal(name, nestedDatum, isNull),
+	];
+};
+/**
+ * gets the animations direction signal events for legends
+ * @param name
+ */
+//TODO: Add tests
+export const getRscLegendColorAnimationDirection = (name: string): { update: string; events: string }[] => {
+	return [
+		{ events: `@${name}_legendEntry:mouseover`, update: '1' },
+		{ events: `@${name}_legendEntry:mouseout`, update: '-1' },
+	];
+};
+/**
+ * gets the animations direction signal events for trend lines
+ * @param name
+ */
+//TODO: Add tests
+export const getRscTrendlineColorAnimationDirection = (name: string): { events: string; update: string }[] => {
+	return [
+		{ events: `@${name}_voronoi:mouseover`, update: '1' },
+		{ events: `@${name}_voronoi:mouseout`, update: '-1' },
+	];
+};
+/**
+ * gets the previous highlighted item events for legends
+ * @param name
+ */
+//TODO: Add tests
+export const getRscLegendHighlightedItemPrev = (name: string): { events: string; update: string }[] => {
+	return [{ events: `@${name}_legendEntry:mouseover`, update: 'null' }];
+};
+/**
+ * gets the animation signal for Opacity animations
+ * @returns Signal
+ */
+//TODO: add tests
+const getRscAnimation = (): Signal => {
+	return {
+		name: RSC_ANIMATION,
+		value: 0,
+		on: [
+			{
+				events: 'timer{16.666666666666668}',
+				update: `scale('rscAnimationCurve', scale('rscAnimationCurveInverse', ${RSC_ANIMATION}) + 0.03333333333333334)`,
+			},
+		],
+	};
+};
+/**
+ * gets the Color animation direction signal for Opacity animations
+ * @param name
+ * @returns Signal
+ */
+//TODO: add tests
+const getRscColorAnimationDirection = (name: string): Signal => {
+	return {
+		name: 'rscColorAnimationDirection',
+		value: -1,
+		on: [
+			{ events: `@${concatName(name)}:mouseover`, update: '1' },
+			{ events: `@${concatName(name)}:mouseout`, update: '-1' },
+		],
+	};
+};
+/**
+ * gets the color animation signal for Opacity animations
+ * @returns Signal
+ */
+//TODO: add tests
+const getRscColorAnimation = (): Signal => {
+	return {
+		name: 'rscColorAnimation',
+		value: 0,
+		on: [
+			{
+				events: 'timer{16.666666666666668}',
```

Comment:
Same as above. A comment explaining why we're using the number, or assigning it to a named variable will make this easier to work with in the future.